design/dedicated-osd-pod.md

```diff
@@ -0,0 +1,73 @@
+# Run OSD in its own Pod
+
+## TL;DR
+
+A one-OSD-per-Pod placement should be implemented to improve reliability and resource efficiency for Ceph OSD daemon.
+
+## Background
+
+Currently in Rook 0.7, Rook Operator starts a ReplicaSet for Rook OSD  on each storage node. The ReplicaSet has just one replica. Rook OSD scans and prepares devices, creates OSD IDs and data directories, generates Ceph configuration. At last, Rook OSD starts all Ceph OSD daemons in foreground and tracks the OSD processes.
+
+As observed, all Ceph OSDs are running in the same Pod.
+
+## Limitations
+
+One Pod for all OSDs doesn't have the highest reliability nor efficiency. If the Pod is deleted, accidentally or during maintenence, all OSDs are down till the ReplicaSet restart. If nodes have asymmetric configurations, some have more devices than others, the Rook Operator doesn't request appropriate resources for the OSD Pods, since the resources are requested before the OSDs are discovered.
+
+A more comprehensive discussions can be found at [this issue](https://github.com/rook/rook/issues/1341).
+
+## Proposal   
+
+We propose the following OSD startup change.
+
+### Create new OSDs
+| Sequence |Rook Operator  | Rook OSD   | Ceph OSD  | Note  |
```

Comment:
We need another column for `Rook Agent`. The agent is already running in the cluster. Its responsibility now will be to detect what devices are available on each node and add the devices to a configmap. We likely want to create one configmap per node to avoid contention between agents. This gives us knowledge about *actual* devices available in the cluster.